client/components/NewEntryForm.js

```diff
@@ -16,27 +15,38 @@ class NewEntryForm extends Component {
     this.handleChange = this.handleChange.bind(this)
     this.handleSubmit = this.handleSubmit.bind(this)
   }
+  async componentDidMount() {
+    const promptId = this.props.user.currentPrompt
+    this.props.getPrompt(promptId)
+  }
   handleChange(event) {
     const {name, value} = event.target
     this.setState({[name]: value})
   }
   handleSubmit(event) {
     event.preventDefault()
     this.props.addEntry({
-      userId: this.props.userId,
+      userId: this.props.user.id,
+      promptId: this.props.user.currentPrompt,
       title: this.state.title,
       content: this.state.content,
     })
+    this.props.updateUser(this.props.user.id, {
+      ...this.props.user,
+      currentPrompt: this.props.user.currentPrompt + 1,
```

Comment:
this is fine but we might want to rethink this. will the prompts always be in numerical order? maybe we have some sort of linked list (OMG) and/or calculate this on the backend after a successful entry submission

Reply:
I agree, I think we will need to rethink prompts altogether... I think this is okay to approve for now though, let me know what you think.